be/src/exec/workgroup/work_group.cpp

```diff
@@ -482,11 +484,15 @@ void WorkGroupManager::apply(const std::vector<TWorkGroupOp>& ops) {
     while (it != _workgroup_expired_versions.end()) {
         auto wg_it = _workgroups.find(*it);
         if (wg_it != _workgroups.end() && wg_it->second->is_removable()) {
-            int128_t wg_id = *it;
+            auto id = wg_it->second->id();
+            auto version = wg_it->second->version();
             _sum_cpu_limit -= wg_it->second->cpu_limit();
             _workgroups.erase(wg_it);
+            if (_workgroup_versions.count(id) && _workgroup_versions.at(id) <= version) {
```

Comment:
the find execute twice.
Why not use find() to get an iterator and erase the iterator too.

Reply:
It is a extremely low-frequent operations.